lib/QoreValue.cpp

```diff
@@ -69,6 +69,10 @@ QoreValue::QoreValue(double f) : type(QV_Float) {
 }
 
 QoreValue::QoreValue(AbstractQoreNode* n) : type(QV_Node) {
+   if (n && n->getType() == NT_NOTHING) {
+      static_cast<QoreNothingNode *>(n)->deref();
```

Comment:
no need to dereference QoreNothingNode - it has no effect - it has the highlander bit set (there_can_be_only_one = true)